client/src/components/QuestionAnswers/Question.jsx

```diff
@@ -61,10 +62,11 @@ function Question({ questionId, questionBody, questionHelpfulness }) {
         </span>
       </div>
       {/* TODO: is this the best place for the modal? */}
-      <AddAnswerModal
+      <Modal
         showModal={showAddAnswer}
         onClose={() => setShowAddAnswer(false)}
-        productData={{ productName, questionId, questionBody }}
+        ModalForm={AddAnswerModal}
+        formData={{ productName, questionId, questionBody }}
```

Comment:
ModalForm and formData !  Cool... I need to look into this for my comparison.

Reply:
Yeah!

Interestingly, this is a Higher-Order Component, like the one that the change request is looking for